src/webgl/p5.RendererGL.js

```diff
@@ -874,12 +879,54 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
     this.curStrokeJoin = join;
   }
 
-  filter(filterType) {
-    // filter can be achieved using custom shaders.
-    // https://github.com/aferriss/p5jsShaderExamples
-    // https://itp-xstory.github.io/p5js-shaders/#/
-    p5._friendlyError('filter() does not work in WEBGL mode');
+  filter(args) {
+    // Couldn't create graphics in RendererGL constructor
+    // (led to infinite loop)
+    // so it's just created here once on the initial filter call.
+    if (!this.filterGraphicsLayer) {
+      this.filterGraphicsLayer =
+        new p5.Graphics(this.width, this.height, constants.WEBGL, this._pInst);
+    }
+    let pg = this.filterGraphicsLayer;
+
+    // TODO, handle different signatures:
+    // if args[0] is a p5.Shader,
+    //   this.filterShader = args[0]
+    // else
+    //   this.filterShader = map(args[0], {GRAYSCALE: grayscaleShader, ...})
+    //   filterOperationParameter = undefined or args[1]
+
+    if (typeof args[0] === 'string') {
```

Comment:
Is this the right check? I think we need to know if args[0] is a p5.shader, or a filter constant ( BLUR, INVERT, etc), and if not, we print an error. Aren't the filter constants strings?

Reply:
Yes, that's the check. Although there shouldn't need to be a third case handling errors, since the args get validated in the parent `filter()` function: [pixels.js#L472](https://github.com/processing/p5.js/blob/3b0d1328b96a6d46dfc77e3f33e97077934967fa/src/image/pixels.js#L472)

Right now the string / filter constant case has an error placeholder. It will be replaced in the next PR that implements the shader versions of filter constants. And the current code handles the p5.Shader case.